fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -367,6 +367,7 @@ private ScoredPlan planFilter(@Nonnull PlanContext planContext, @Nonnull QueryCo
                 //   * size of row.
                 //   * need for type filtering if row scan with multiple types.
                 if (bestPlan == null || p.score > bestPlan.score ||
+                        p.unsatisfiedFilters.size() < bestPlan.unsatisfiedFilters.size() ||
```

Comment:
Without this change, there is actually a rank query that changed. In particular, a rank query with a single grouping field and a single ranked field turned from "scan the rank index over the right range" to "scan a different rank index over the range and then filter records by the grouping column". This is a sign that the change could have knock on side effects.